catroid/src/main/java/org/catrobat/catroid/content/bricks/PlaySoundAndWaitBrick.java

```diff
@@ -0,0 +1,376 @@
+
+		protected Context context;
+		protected ArrayAdapter<SoundInfo> spinnerAdapter;
+
+		private boolean isTouchInDropDownView;
+
+		public SpinnerAdapterWrapper(Context context, ArrayAdapter<SoundInfo> spinnerAdapter) {
+			this.context = context;
+			this.spinnerAdapter = spinnerAdapter;
+
+			this.isTouchInDropDownView = false;
+		}
+
+		@Override
+		public void registerDataSetObserver(DataSetObserver paramDataSetObserver) {
+			spinnerAdapter.registerDataSetObserver(paramDataSetObserver);
+		}
+
+		@Override
+		public void unregisterDataSetObserver(DataSetObserver paramDataSetObserver) {
+			spinnerAdapter.unregisterDataSetObserver(paramDataSetObserver);
+		}
+
+		@Override
+		public int getCount() {
+			return spinnerAdapter.getCount();
+		}
+
+		@Override
+		public Object getItem(int paramInt) {
+			return spinnerAdapter.getItem(paramInt);
+		}
+
+		@Override
+		public long getItemId(int paramInt) {
+			SoundInfo currentSound = spinnerAdapter.getItem(paramInt);
+			if (!currentSound.getTitle().equals(context.getString(R.string.new_broadcast_message))) {
+				oldSelectedSound = currentSound;
+			}
+			return spinnerAdapter.getItemId(paramInt);
+		}
+
+		@Override
+		public boolean hasStableIds() {
+			return spinnerAdapter.hasStableIds();
+		}
+
+		@Override
+		public View getView(int paramInt, View paramView, ViewGroup paramViewGroup) {
+			if (isTouchInDropDownView) {
+				isTouchInDropDownView = false;
+				if (paramInt == 0) {
+					switchToSoundFragmentFromScriptFragment();
+				}
+			}
+			return spinnerAdapter.getView(paramInt, paramView, paramViewGroup);
+		}
+
+		@Override
+		public int getItemViewType(int paramInt) {
+			return spinnerAdapter.getItemViewType(paramInt);
+		}
+
+		@Override
+		public int getViewTypeCount() {
+			return spinnerAdapter.getViewTypeCount();
+		}
+
+		@Override
+		public boolean isEmpty() {
+			return spinnerAdapter.isEmpty();
+		}
+
+		@Override
+		public View getDropDownView(int paramInt, View paramView, ViewGroup paramViewGroup) {
+			View dropDownView = spinnerAdapter.getDropDownView(paramInt, paramView, paramViewGroup);
+
+			dropDownView.setOnTouchListener(new OnTouchListener() {
+				@Override
+				public boolean onTouch(View paramView, MotionEvent paramMotionEvent) {
+					isTouchInDropDownView = true;
+					return false;
+				}
+			});
+
+			return dropDownView;
+		}
+
+		private void switchToSoundFragmentFromScriptFragment() {
+			ScriptActivity scriptActivity = ((ScriptActivity) context);
+			scriptActivity.switchToFragmentFromScriptFragment(ScriptActivity.FRAGMENT_SOUNDS);
+
+			setOnSoundInfoListChangedAfterNewListener(context);
+		}
+	}
+
+	@Override
+	public void onSoundInfoListChangedAfterNew(SoundInfo soundInfo) {
+		sound = soundInfo;
+		oldSelectedSound = soundInfo;
+	}
+
+	@Override
+	public void storeDataForBackPack(Sprite sprite) {
+		SoundInfo backPackedSoundInfo = SoundController.getInstance().backPackHiddenSound(this.getSound());
```

Comment:
Please just use     sound    instead of this.getSound() - this is a bit more compact